codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/http/HttpBindingGenerator.kt

```diff
@@ -146,9 +147,14 @@ class HttpBindingGenerator(
                 deserializeFromHeader(model.expectShape(target.value.target), binding.member)
             }
         }
+        val retType = if (outputT.isOptional()) {
+            "#T"
+        } else {
+            "std::option::Option<#T>"
+        }
```

Comment:
This is correct.

It's also equivalent to:

```
val retType = outputT.mapRustType { it.asOptional() }.rustType().render()
```

However, I think it's best to deal with symbols and have the `CodeWriter` always render them as strings when possible.

```
val returnTypeSymbol = outputT.mapRustType { it.asOptional() }
```

And then below do:

```
            writer.rustBlock(
                "pub fn $fnName(header_map: &#T::HeaderMap) -> std::result::Result<#T, #T::ParseError>",
                RuntimeType.http,
                returnTypeSymbol,
                headerUtil
            ) {
```

nit: now that we're working on this, I'd rename `outputT` as `outputSymbol`, since it's a `Symbol`, not a `RustType`.